frontend/src/metabase/admin/permissions/utils/graph/partial-updates.ts

```diff
@@ -34,19 +34,27 @@ export function getModifiedGroupsPermissionsGraphParts(
 export function mergeGroupsPermissionsUpdates(
   originalDataPermissions: GroupsPermissions | null | undefined,
   newDataPermissions: GroupsPermissions,
+  modifiedGroupIds: string[],
```

Comment:
I noticed an issue where setting values to blocked removed all permissions for a group causing the API to omit the group's graph from the response if it only used default values for all databases in that group (a weird but valid API design decision we're choosing to live with). The FE assumes an empty object is default permissions, so I'm detecting that case and merging in the empty record.

Should fix #46170, but need to double check if that completely solves all issues in that ticket.